src/linear_algebra/invariant_submodule.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import linear_algebra.basic
+import linear_algebra.projection
+
+/-!
+# Invariant submodules
+
+In this file, we define and prove some basic results on invariant submodules.
+-/
+
+namespace submodule
+
+variables {E R : Type*} [ring R] [add_comm_group E] [module R E]
+
+/-- `U` is `T` invariant (ver 1): `U ≤ U.comap` -/
+def invariant_under (U : submodule R E) (T : E →ₗ[R] E) : Prop := U ≤ U.comap T
+
+/-- `U` is `T` invariant if and only if `U.map T ≤ U` -/
+@[simp] lemma invariant_under_iff_map (U : submodule R E) (T : E →ₗ[R] E) :
+  U.invariant_under T ↔ U.map T ≤ U := submodule.map_le_iff_le_comap.symm
+
+/-- `U` is `T` invariant if and only if `set.maps_to T U U` -/
+lemma invariant_under_iff_maps_to (U : submodule R E) (T : E →ₗ[R] E) :
+  set.maps_to T U U ↔ U.invariant_under T := iff.rfl
```

Comment:
Lemmas called `A_iff_B` should be stated as `A ↔ B`, but here you have `B ↔ A`. I think you should keep the order in the name, but flip the direction.